workers/data_refinery_workers/processors/smasher.py

```diff
@@ -0,0 +1,254 @@
+            i = 1
+            while i < len(all_frames):
+                merged = merged.merge(all_frames[i], left_index=True, right_index=True)
+                i = i + 1
+
+            job_context['merged'] = merged
+
+            # Transpose before scaling
+            transposed = merged.transpose()
+
+            # Scaler
+            if job_context['dataset'].scale_by != "NONE":
+                scale_funtion = scalers[job_context['dataset'].scale_by]
+                scaler = scale_funtion(copy=True)
+                scaler.fit(transposed)
+                scaled = pd.DataFrame(  scaler.transform(transposed), 
+                                        index=transposed.index, 
+                                        columns=transposed.columns
+                                    )
+                # Untranspose
+                untransposed = scaled.transpose()
+            else:
+                # Wheeeeeeeeeee
+                untransposed = transposed.transpose()
+
+            job_context['final_frame'] = untransposed
+            
+            # Write to temp file with dataset UUID in filename.
+            outfile = smash_path + key + ".tsv"
+            untransposed.to_csv(outfile, sep='\t', encoding='utf-8')
+
+        # Copy LICENSE.txt and README.md files
+        shutil.copy("README_DATASET.md", smash_path + "README.md")
+        shutil.copy("LICENSE_DATASET.txt", smash_path + "LICENSE.TXT")
+
+        # Create metadata file.
+        metadata = {}
+        metadata['files'] = os.listdir(smash_path)
+        metadata['num_samples'] = num_samples
+        metadata['num_experiments'] = job_context["experiments"].count()
+        metadata['aggregate_by'] = job_context["dataset"].aggregate_by
+        metadata['scale_by'] = job_context["dataset"].scale_by
+
+        samples = {}
+        for sample in job_context["dataset"].get_samples():
+            samples[sample.title] = sample.to_metadata_dict()
+        metadata['samples'] = samples
+
+        experiments = {}
+        for experiment in job_context["dataset"].get_experiments():
+            experiments[experiment.accession_code] = experiment.to_metadata_dict()
+        metadata['experiments'] = experiments
+
+        metadata['created_at'] = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S')
+        with open(smash_path + 'metadata.json', 'w') as metadata_file: 
+            json.dump(metadata, metadata_file, indent=4, sort_keys=True)
+
+        # Finally, compress all files into a zip
+        final_zip = smash_path + str(job_context['dataset'].id)
+        shutil.make_archive(final_zip, 'zip', smash_path)
+        job_context["output_file"] = final_zip + ".zip"
+    except Exception as e:
+        job_context['dataset'].success = False
+        job_context['dataset'].failure_reason = "Failure reason: " + str(e)
+        job_context['dataset'].save()
+        job_context['success'] = False
+        job_context['failure_reason'] = str(e)
+        return job_context
```

Comment:
If this fails, we will never notify the user?

Reply:
That behavior has not been described yet. I don't know what we want to do there.

Reply:
Do we have an issue or anything to track it?